_posts/2022-XX-XX-configuring-git-repositories.adoc

```diff
@@ -0,0 +1,269 @@
+image::/assets/img/configuring-git-repositories/blog-workspace.png[Working on the blog repository files in Che-Theia]
+{empty}
+
+=== Contributing to the remote Git repository from the workspace
+
+https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token[GitHub], https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html[GitLab], and https://confluence.atlassian.com/bitbucketserver/personal-access-tokens-939515499.html[Bitbucket] authentication is based on using personal access tokens (PAT). Each user is able to request several PATs with different names, permissions, expiration times, and so on. Those tokens can be used to sign REST API calls and perform Git repository operations. {prod-short} requires a PAT in a Kubernetes secret for starting a new workspace with a clone of a private Git repository in it. From this secret, another secret is automatically generated, allowing authorization for Git operations. The automatically generated secret is mounted into a workspace container as a Git credentials file. You could request a PAT from a Git SCM server and create the secret yourself; however, the preferred method is to use OAuth. With OAuth, {prod-short} can request an access token on your behalf and manage its renewal.
+
+To authenticate my Git operations to a Git SCM server (such as pushing to a remote branch), I need the Git credentials file. I explain how to set up the OAuth authorization in <<Accessing a private Git repository>>.
+
+Now that I can view my project's source code in the Che-Theia IDE, I'll go ahead and make some change in the project source code. I'm creating a new branch called `test` and adding a few empty lines to the `README.md` file.
+
+.Source control options in Che-Theia
+image::/assets/img/configuring-git-repositories/source-control.png[Source control in Che-Theia]
+{empty}
+
+I must also set my Git user name and user email (if I haven't done so already) that will identify my commits.
+
+.Configuring Git user name and user email in Che-Theia
+image::/assets/img/configuring-git-repositories/git-config.png[Git config in Che-Theia]
+{empty}
+
+ I commit my changes and push them to the remote Git repository.
+
+.Using the Push option in Che-Theia to push to the test branch
+image::/assets/img/configuring-git-repositories/source-control.png[Using the Push option in Che-Theia]
+{empty}
+
+From here, I can open a pull request to the https://github.com/eclipse-che/blog[eclipse-che/blog] repository.
+
+
+=== Accessing a private Git repository
+
+Before you can access a private Git repository, some actions are required from you as a non-privileged user and from your {prod-short} administrator. Those actions are required for configuring the OAuth authorization so that {prod-short} can act on behalf of the user account on the Git SCM server. GitHub and GitLab use OAuth2, whereas Bitbucket uses OAuth1.
+
+==== User side
```

Comment:
Hello, according to the [docs](https://www.eclipse.org/che/docs/che-7/administration-guide/configuring-authorization/#configuring-github-oauth_che) and this comment: https://github.com/eclipse/che/issues/21298#issuecomment-1079020296  this should be an task meant to be performed by the admin.

Could we remove this `User side` section since the oauth app creation, retrieving client ID and secret should be also done by an admin?